### PR TITLE
Disallow navigation to and blocking of anonymous customers

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

**Comment by josepfo** [comment]
[Disallow navigation to, and blocking of, anonymous customer](https://trello.com/c/Y5sXv5V8/616-disallow-navigation-to-and-blocking-of-anonymous-customer)

**Review by hellypi — COMMENTED** [review]
.

File: lib/new-admin/graphql/schema.js [hunk 1]
```diff
@@ -402,7 +409,8 @@ const resolvers = {
     resetAccounts: (...[, { schemaVersion }]) => settingsLoader.resetAccounts(schemaVersion),
     setCustomer: (root, args, context, info) => {
       const token = context.req.cookies && context.req.cookies.token
-      return customers.updateCustomer(args.customerId, args.customerInput, token)
+      if (args.customerId !== anonymous.uuid) return customers.updateCustomer(args.customerId, args.customerInput, token)
+      return customers.getCustomerById(args.customerId)
```
Comment (RafaelTaranto) [inline lib/new-admin/graphql/schema.js]: It's cleaner if you do the other way around

```suggestion
      if (args.customerId === anonymous.uuid) return customers.getCustomerById(args.customerId)
      return customers.updateCustomer(args.customerId, args.customerInput, token)
      
```

File: lib/new-admin/graphql/schema.js [hunk 2]
```diff
@@ -402,7 +409,8 @@ const resolvers = {
     resetAccounts: (...[, { schemaVersion }]) => settingsLoader.resetAccounts(schemaVersion),
     setCustomer: (root, args, context, info) => {
```
Comment (RafaelTaranto) [inline lib/new-admin/graphql/schema.js]: Also might as well desconstruct args here.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**